### PR TITLE
Fix getting pointed node

### PR DESCRIPTION
**Pull request by RealBadAngel**
Fixes issue #3719

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -382,12 +382,19 @@ PointedThing getPointedThing(Client *client, Hud *hud, const v3f &player_positio
 					box.MinEdge += npf;
 					box.MaxEdge += npf;
 
+					v3f centerpoint = box.getCenter();
+					f32 distance = (centerpoint - camera_position).getLength();
+					if (distance >= mindistance)
+							continue;
+					if (!box.intersectsWithLine(shootline))
+							continue;
```
Comment (nerzhul) [inline src/game.cpp]: one less tab

File: src/game.cpp [hunk 2]
```diff
@@ -382,12 +382,19 @@ PointedThing getPointedThing(Client *client, Hud *hud, const v3f &player_positio
 					box.MinEdge += npf;
 					box.MaxEdge += npf;
 
+					v3f centerpoint = box.getCenter();
+					f32 distance = (centerpoint - camera_position).getLength();
+					if (distance >= mindistance)
+							continue;
```
Comment (nerzhul) [inline src/game.cpp]: one less tab

**Comment by RealBadAngel** [comment]
Ok, ive reworked completely the search for pointed node, should become way faster than it was before.
Check for above surface node is done only once if pointed under is found.

**Comment by Fixer-007** [comment]
Playing with it, spotted no problems with it at the moment.

**Comment by Fixer-007** [comment]
Fixed issue above.

**Comment by RealBadAngel** [comment]
https://github.com/RealBadAngel/minetestHD/commit/5c6ee4aea1d358fa9d7c76cea9c2b3abc72657d4
this adds highlighting of selected face (and depends on previous commit)
screenshot_20160221_061731
screenshot_20160221_061736

**Comment by paramat** [comment]
I oppose the 2nd commit, it's clever and pretty but i feel not really needed.

**Comment by kilbith** [comment]
Agree with paramat, this is just a distracting visual gadget.

**Comment by nerzhul** [comment]
You face enhancement is pretty, i like it but it should not be in this PR

**Comment by RealBadAngel** [comment]
@nerzhul @paramat then just merge actual bugfix and close it.

**Comment by est31** [comment]
:+1: to the first commit.

**Comment by paramat** [comment]
:+1: First commit.

**Pull request closed without merging**